File: 2021/day-1/part-2/main.py
#!/usr/bin/env python3

import sys

def read_input(filename):
    input = []
    with open(filename) as fd:
        lines = fd.readlines()
        for line in lines:
            input.append(int(line.strip()))
    return input

def main():
    input = read_input("input.txt")
    result = 0
    for i in range(0, len(input)-3):
        lower_sum = sum(input[i:i+3])
        upper_sum = sum(input[i+1:i+4])
        if upper_sum > lower_sum:
            result += 1
    print(result)
    
if __name__ == "__main__":
    sys.exit(main())

File: 2021/day-1/part-2/main.jl
function main()
    local result = 0
    local content = readlines("input.txt")
    local lines = parse.(Int, content)
    for i in 4:length(lines)
        local lower_sum = sum(lines[i-3:i-1])
        local upper_sum = sum(lines[i-2:i])
        if upper_sum > lower_sum
            result += 1
        end
    end
    println(result)
end

main()